control-plane/p4RuntimeSerializer.cpp

```diff
@@ -766,13 +840,29 @@ class P4RuntimeAnalyzer {
             addDocumentation(param, actionParam->to<IR::IAnnotated>());
 
             auto paramType = typeMap->getType(actionParam, true);
-            if (!paramType->is<IR::Type_Bits>() && !paramType->is<IR::Type_Boolean>()) {
-                ::error("Action parameter %1% has a type which is not bit<> or int<> or bool",
-                        actionParam);
+            if (!paramType->is<IR::Type_Bits>() && !paramType->is<IR::Type_Boolean>()
+                && !paramType->is<IR::Type_Newtype>() &&
+                !paramType->is<IR::Type_SerEnum>()) {
+                ::error("Action parameter %1% has a type which is not "
+                        "bit<>, int<>, bool, type or Serializable Enum", actionParam);
                 continue;
             }
             if (paramType->is<IR::Type_Boolean>()) {
                 param->set_bitwidth(1);
+            } else if (paramType->is<IR::Type_Newtype>()) {
+                auto w = getTypeWidth(paramType, typeMap, refMap);
+                param->set_bitwidth(w);
+                cstring type_name = getTypeName(paramType);
+                if (type_name) {
+                    auto namedType = new p4configv1::P4NamedType();
+                    namedType->set_name(type_name);
+                    param->set_allocated_type_name(namedType);
```

Comment:
I commented on this in another place already. Please read the Protobuf documentation on when to use the `set_allocated_*` field setters. It is almost never required, and certainly not in this case.
```
auto namedType = param->mutable_type_name();
namedType->set_name(type_name);
```